src/libshadertrap/src/parser.cc

```diff
@@ -257,15 +258,59 @@ bool Parser::ParseCommandAssertEqual() {
             }}},
           // BUFFERS and RENDERBUFFERS are mutually exclusive parameters
           {{Token::Type::kKeywordBuffers,
-            Token::Type::kKeywordRenderbuffers}})) {
+            Token::Type::kKeywordRenderbuffers}},
+
+          {{Token::Type::kKeywordFormat, [this, &format_entries]() -> bool {
+        while (true) {
+          CommandAssertEqual::FormatEntry::Kind kind;
+          switch (tokenizer_->PeekNextToken()->GetType()) {
+            case Token::Type::kKeywordSkipBytes:
+              kind = CommandAssertEqual::FormatEntry::Kind::kSkip;
+              break;
+            case Token::Type::kKeywordTypeByte:
+              kind = CommandAssertEqual::FormatEntry::Kind::kByte;
+              break;
+            case Token::Type::kKeywordTypeFloat:
+              kind = CommandAssertEqual::FormatEntry::Kind::kFloat;
+              break;
+            case Token::Type::kKeywordTypeInt:
+              kind = CommandAssertEqual::FormatEntry::Kind::kInt;
+              break;
+            case Token::Type::kKeywordTypeUint: {
+              kind = CommandAssertEqual::FormatEntry::Kind::kUint;
+              break;
+              case Token::Type::kString:
+                kind = CommandAssertEqual::FormatEntry::Kind::kString;
+                break;
+              default:
+                return true;
+            }
+          }
+          auto format_start_token = tokenizer_->NextToken();
+          size_t count;
+          if (kind == CommandAssertEqual::FormatEntry::Kind::kString) {
+            count = 0;
+          } else {
+            auto maybe_count = ParseUint32("count");
+            if (!maybe_count.first) {
+              return false;
+            }
+            count = maybe_count.second;
+          }
+          format_entries.push_back(
+              {std::move(format_start_token), kind, count});
+        }
+      }}})) {
     return false;
   }
   parsed_commands_.push_back(MakeUnique<CommandAssertEqual>(
```

Comment:
At this point you could check whether `arguments_are_renderbuffers` is true.

If it is, you could check that `format_entries` is empty, and give an error if it is non-empty.

If `arguments_are_renderbuffers` is false but `format_entries` is empty, you could set `format_entries` to contain a single byte format entry matching the length of the first buffer.